### ISSUE TITLE
Raytracing doesn't work in a function

### ISSUE THREAD

**Issue by eem50** [ShaneBeee/SkBee#341]
Whenever I let the player ray trace in a function, it just doesn't work.

I use this code to test it:
```vb
command /testraytracefunction:
	trigger:
		broadcast "1"
		testFunction(player)
		
function testFunction(p: player):
	set {_ray} to ray trace from {_p} with max distance 25
	set {_e} to ray trace hit entity of {_ray}
	if {_e} is set:
		broadcast "You have hit an entity"
	else:
		broadcast "You haven't hit anything"
	
command /testraytracecommand:
	trigger:
		broadcast "2"
		set {_ray} to ray trace from player with max distance 25
		set {_e} to ray trace hit entity of {_ray}
		if {_e} is set:
			broadcast "You have hit an entity"
		else:
			broadcast "You haven't hit anything"
```

I have made two commands to compare what I expect it to do, along what it does inside a function. I have screenshots of executing them both, while targeting a cow:

![2023-02-11_20 46 57](https://user-images.githubusercontent.com/96268837/218278248-cf4a2ebb-aed1-44e6-9bc8-ff65f2b19e48.png)

![2023-02-11_20 45 59](https://user-images.githubusercontent.com/96268837/218278234-e88e28f9-0cee-4074-85b4-3b81d0839f26.png)

I hope I explained the problem well enough, if not I can always hand more information!

Version:

![2023-02-11_20 42 30](https://user-images.githubusercontent.com/96268837/218278316-ebc7ccef-aaed-4ec6-9d1b-6d4355570a91.png)

**Comment by ShaneBeee** [comment]
testing your function I get this in console
```
[11:50:50 ERROR]: #!#! Stack trace:
[11:50:50 ERROR]: #!#! java.lang.ClassCastException: class [Lorg.bukkit.entity.Entity; cannot be cast to class [Lorg.bukkit.entity.LivingEntity; ([Lorg.bukkit.entity.Entity; and [Lorg.bukkit.entity.LivingEntity; are in unnamed module of loader java.net.URLClassLoader @1794d431)
[11:50:50 ERROR]: #!#!     at SkBee-2.7.5.jar//com.shanebeestudios.skbee.elements.raytrace.expressions.ExprRayTraceFromEntity.get(ExprRayTraceFromEntity.java:72)
[11:50:50 ERROR]: #!#!     at SkBee-2.7.5.jar//com.shanebeestudios.skbee.elements.raytrace.expressions.ExprRayTraceFromEntity.get(ExprRayTraceFromEntity.java:27)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.util.SimpleExpression.getArray(SimpleExpression.java:99)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.effects.EffChange.execute(EffChange.java:274)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:61)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:56)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.function.ScriptFunction.execute(ScriptFunction.java:85)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.function.Function.execute(Function.java:129)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.function.FunctionReference.execute(FunctionReference.java:299)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.function.EffFunctionCall.execute(EffFunctionCall.java:52)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:61)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:56)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.ScriptCommand.execute2(ScriptCommand.java:286)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.ScriptCommand.lambda$execute$0(ScriptCommand.java:251)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.ScriptCommand.execute(ScriptCommand.java:256)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.Commands.handleCommand(Commands.java:261)
[11:50:50 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.Commands$1.onPlayerCommand(Commands.java:178)
[11:50:50 ERROR]: #!#!     at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40)
[11:50:50 ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[11:50:50 ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[11:50:50 ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:672)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2264)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$20(PlayerConnection.java:2248)
[11:50:50 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59)
[11:50:50 ERROR]: #!#!     at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.TickTask.run(TickTask.java:18)
[11:50:50 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:153)
[11:50:50 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1341)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:197)
[11:50:50 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:126)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.bh(MinecraftServer.java:1318)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1311)
[11:50:50 ERROR]: #!#!     at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:136)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.i_(MinecraftServer.java:1289)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1177)
[11:50:50 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
[11:50:50 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:1589)
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! Version Information:
[11:50:50 ERROR]: #!#!   Skript: 2.6.4 (latest)
[11:50:50 ERROR]: #!#!     Flavor: skriptlang-github
[11:50:50 ERROR]: #!#!     Date: 17:14:25.041218900
[11:50:50 ERROR]: #!#!   Bukkit: 1.19.3-R0.1-SNAPSHOT
[11:50:50 ERROR]: #!#!   Minecraft: 1.19.3
[11:50:50 ERROR]: #!#!   Java: 19.0.1 (OpenJDK 64-Bit Server VM 19.0.1+10)
[11:50:50 ERROR]: #!#!   OS: Mac OS X aarch64 13.1
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! Server platform: Paper
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! Current node: null
[11:50:50 ERROR]: #!#! Current item: set {_ray} (as java.lang.Object) to ray trace from {_p} (as org.bukkit.entity.Entity) with max distance [[long:25]]
[11:50:50 ERROR]: #!#! Current trigger: function testFunction (simple event) (test.sk, line 7)
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! Thread: Server thread
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! Language: english
[11:50:50 ERROR]: #!#! Link parse mode: LENIENT
[11:50:50 ERROR]: #!#! 
[11:50:50 ERROR]: #!#! End of Error.
[11:50:50 ERROR]: #!#! 

```
so yes, definitely a bug

**Comment by eem50** [comment]
Wow thanks for the fast response! Will you be able to fix it?

**Comment by ShaneBeee** [comment]
definitely, working on a fix right now :)

**Comment by eem50** [comment]
Ah great! GL :D

**Comment by eem50** [comment]
Hey I downloaded the latest version (2.7.5) and gave the same skript a try and still, it didn't work. Idk, maybe its the skript itself? Im using the same code as earlier and get the same results.

**Comment by ShaneBeee** [comment]
I haven't released this yet, you reported this issue after I released SkBee 2.7.5

**Comment by eem50** [comment]
Ooooh nevermind! Stupid of me to think :P I'll try it again whenever you release the next version.
